neo4j/internal/bolt/bolt4.go

```diff
@@ -109,17 +109,12 @@ func NewBolt4(serverName string, conn net.Conn, log log.Logger) *bolt4 {
 		packer:  &packstream.Packer{},
 		onErr:   func(err error) { b.setError(err, true) },
 	}
-	// Setup open streams. Errors reported to callback are assertion like errors, let them
-	// bubble up to kill them off when they happen, alternatively they could be logged.
-	b.streams.onAssertFail = func(err error) {
-		b.setError(err, false)
-	}
-	b.streams.onClose = func(s *stream) { // Current stream succesfully closed (no error)
-		if len(s.sum.Bookmark) > 0 {
-			b.bookmark = s.sum.Bookmark
-		}
-	}
-	b.streams.onEmpty = func() { // All streams closed (succesfully or with error)
+
+	return b
+}
+
+func (b *bolt4) checkStreams() {
+	if b.streams.num <= 0 {
```

Comment:
can it be strictly lower than 0?

Reply:
Not really... it Is an error that might be worth logging as debug